File: 0485-Max Consecutive Ones/golang/solution.go
package golang

func findMaxConsecutiveOnes(nums []int) int {
	var maxConsecutive int
	var currConsecutive int
	length := len(nums)
	for i := 0; i < length; i++ {
		if nums[i] == 1 {
			currConsecutive++
		} else {
			currConsecutive = 0
		}

		if currConsecutive > maxConsecutive {
			maxConsecutive = currConsecutive
		}
	}

	return maxConsecutive
}

// Runtime: 36 ms, faster than 96.94% of Go online submissions for Max Consecutive Ones.
// Memory Usage: 6.4 MB, less than 96.94% of Go online submissions for Max Consecutive Ones.

File: 0485-Max Consecutive Ones/c++/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        ios_base :: sync_with_stdio(false);
        cin.tie(0);
        cout.tie(0);
        
        int length = nums.size();
        int solution = 0;
        
        for(int i = 0; i < length; ++i) {
            int count = 0;
            while(i < length && nums[i]==1) {
                count++;
                i++;
            }
            if(count > solution) solution = count;
        }
        
        return solution;
    }
};

// Runtime: 60 ms, faster than 99.88% of C++ online submissions for Max Consecutive Ones.
// Memory Usage: 36.3 MB, less than 48.85% of C++ online submissions for Max Consecutive Ones. 